### DISCUSSION TITLE
How to properly terminate an idle connection (xrdp + xvnc)?

### DISCUSSION THREAD

**Discussion by rayrapetyan** [neutrinolabs/xrdp#2096] [Q&A] [answered]
Hello, I'm using xvnc as a "last mile" protocol (due to https://github.com/neutrinolabs/xrdp/issues/2046 bug), and wondering how to properly kill an idle connection in this case? I've tried `-IdleTimeout` xvnc option, it works, but in a weird way - connection terminates after timeout, but  when I reconnect - I see everything remain the same: rdp connection remains alive.

**Comment by matt335672** [comment, marked as answer]
Hi @rayrapetyan 

The `-IdleTimeout` option on xvnc removes VNC clients (in this case the xrdp process), but leaves the session running. From the TigerVNC manpage:-

>       -IdleTimeout seconds
 >             The  number  of seconds after which an idle VNC connection will be dropped.  Default is 0, which means that idle connec‐
 >             tions will never be dropped.



You possibly want to use `-MaxIdleTime` instead:-

>       -MaxIdleTime seconds
>              Terminate after N seconds of user inactivity.  Default is 0.

Does that do what you want?

**Reply by rayrapetyan** [reply]
Thanks! Missed that parameter...
